### PR TITLE
Merging latest Typeset to fork.

### PR DESCRIPTION
**Pull request by squalx**

### CODE REVIEW

**Comment by squalx** [comment]
Sorry about that. I still don't understand Github PRs very well, I have experience with BitBucket and Stash.

Anyway, I added underline and baseline to Typeset.

**Comment by squalx** [comment]
By the way, I am using these features already. Have been using them for a while in an App I'm building.

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,7 @@ DerivedData
 *.hmap
 *.ipa
 *.xcuserstate
+.DS_Store
```
Comment (squalx) [inline .gitignore:19]: This is to avoid tracking these hidden Mac files.

File: Classes/NSString+Typeset.h [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 @class TypesetKit;
 
+#define Typeset(__X__) (__X__?__X__.typeset:@"".typeset)
```
Comment (squalx) [inline Classes/NSString+Typeset.h:13]: Mmmm. Maybe this comes from a fork merge, but this code is already in your repo.

File: Classes/TypesetKit.h [hunk 3]
```diff
@@ -15,6 +15,8 @@ typedef TypesetKit *(^TypesettingCGFloatBlock)(CGFloat);
 typedef TypesetKit *(^TypesettingRangeBlock)(NSRange);
 typedef TypesetKit *(^TypesettingStringBlock)(NSString *);
 typedef TypesetKit *(^TypesettingColorBlock)(UIColor *);
+typedef TypesetKit *(^TypesettingBaselineBlock)(int);
```
Comment (squalx) [inline Classes/TypesetKit.h]: Using ints instead of NSNumber to simplify the methods / functions.

File: Classes/TypesetKit.m [hunk 4]
```diff
@@ -156,6 +156,27 @@ - (TypesettingIntegerBlock)hexColor {
...
+        return self;
+    };
+}
+
+
+- (TypesettingStrikeThroughBlock)strikeThrough {
+    return ^(int baseline) {
+        for (NSValue *value in self.attributeRanges) {
+            NSRange range = [value rangeValue];
+            [self.string addAttribute:NSStrikethroughStyleAttributeName value:@(baseline) range:range];
```
Comment (squalx) [inline Classes/TypesetKit.m]: Notice these attributes use NSNumbers. They're being converted from an int.

File: .gitignore [hunk 5]
```diff
@@ -16,6 +16,9 @@ DerivedData
...
+.idea
```
Comment (squalx) [inline .gitignore:20]: This is an AppCode folder. Not sure if anyone opposes to this one.

**Comment by draveness** [comment]
I think use `NSInteger` instead of `int` is a goog choice.

File: Classes/TypesetKit.h [hunk 6]
```diff
@@ -10,11 +10,29 @@
...
+
+// TSStrikeThrough is just a copy of TSUnderline
+typedef TSUnderline TSStrikeThrough;
+
 typedef TypesetKit *(^TypesettingIntegerBlock)(NSUInteger);
 typedef TypesetKit *(^TypesettingCGFloatBlock)(CGFloat);
 typedef TypesetKit *(^TypesettingRangeBlock)(NSRange);
 typedef TypesetKit *(^TypesettingStringBlock)(NSString *);
 typedef TypesetKit *(^TypesettingColorBlock)(UIColor *);
+typedef TypesetKit *(^TypesettingBaselineBlock)(NSInteger);
```
Comment (squalx) [inline Classes/TypesetKit.h:34]: Made this an NSInteger

File: Classes/TypesetKit.h [hunk 7]
```diff
@@ -10,11 +10,29 @@
...
+typedef TypesetKit *(^TypesettingStrikeThroughBlock)(TSStrikeThrough);
```
Comment (squalx) [inline Classes/TypesetKit.h:35]: Using new Typedef to be able to understand the settings.

File: Classes/TypesetKit.h [hunk 8]
```diff
@@ -10,11 +10,29 @@
 
 @interface TypesetKit : NSObject
 
+typedef enum {
```
Comment (squalx) [inline Classes/TypesetKit.h:13]: Added this enum which would be good to use on underline, but I'm using on strikethrough. That way the user can understand the settings for the strikethrough.

File: Typeset/ViewController.m [hunk 9]
```diff
@@ -19,13 +19,19 @@ - (void)viewDidLoad {
     [super viewDidLoad];
     UILabel *label = [[UILabel alloc] initWithFrame:self.view.frame];
     [self.view addSubview:label];
+    label.numberOfLines = 0;
+    
     label.textAlignment = NSTextAlignmentCenter;
-    NSMutableAttributedString *mas = @"Hello typeset, hello".typeset.matchAll(@"Hello").fontSize(40)
```
Comment (squalx) [inline Typeset/ViewController.m:23]: Added more examples of use.
typeset-example

**Comment by draveness** [comment]
Thanks for your great work

**Comment by squalx** [comment]
Thanks for this great library. ;-)

**Pull request merged**